client/landing/gutenboarding/onboarding-block/vertical-background.tsx

```diff
@@ -14,53 +21,56 @@ export interface VerticalBackgroundProps {
 	onLoad: () => void;
 }
 
-const VerticalBackground: FunctionComponent< VerticalBackgroundProps > = ( { id, onLoad } ) => {
-	const [ imageUrl, setImageUrl ] = useState< string | null >( null );
+const VerticalBackground: FunctionComponent< VerticalBackgroundProps > = ( { onLoad } ) => {
+	const { siteVertical } = useSelect( select => select( STORE_KEY ).getState() );
+	const [ imageUrl, setImageUrl ] = useState< string | undefined >();
 
 	// Prefetch the default image.
 	useEffect( () => void ( new window.Image().src = defaultImageUrl ), [] );
 
 	useEffect( () => {
-		const preloadImage = new window.Image();
-		const failureHandler = () => {
-			setImageUrl( defaultImageUrl );
-			onLoad();
-		};
-		// We get an esmodule wrapping the url here
-		const successHandler = ( { default: url }: { default: string } ) => {
-			preloadImage;
-			preloadImage.onload = () => {
-				setImageUrl( url );
+		if ( siteVertical ) {
+			const preloadImage = new window.Image();
+			const failureHandler = () => {
+				setImageUrl( defaultImageUrl );
 				onLoad();
 			};
-			preloadImage.onerror = failureHandler;
-			preloadImage.src = url;
-		};
-		import(
-			/**
-			 * Set eager mode because these are just image _paths_.
-			 */
-			/* webpackMode: "eager" */
-			/* webpackInclude: /\.jpg$/ */
-			`../../../../static/images/verticals/${ id }.jpg`
-		).then( successHandler, failureHandler );
-		return () => {
-			preloadImage.onload = null;
-			preloadImage.onerror = null;
-		};
-	}, [ id, onLoad ] );
+			// We get an esmodule wrapping the url here
+			const successHandler = ( { default: url }: { default: string } ) => {
+				preloadImage.onload = () => {
+					setImageUrl( url );
+					onLoad();
+				};
+				preloadImage.onerror = failureHandler;
+				preloadImage.src = url;
+			};
+			import(
+				/**
+				 * Set eager mode because these are just image _paths_.
+				 */
+				/* webpackMode: "eager" */
+				/* webpackInclude: /\.jpg$/ */
+				`../../../../static/images/verticals/${ siteVertical.id }.jpg`
+			).then( successHandler, failureHandler );
+			return () => {
+				preloadImage.onload = null;
+				preloadImage.onerror = null;
+			};
+		}
+		return setImageUrl( undefined );
```

Comment:
There shouldn't be a `return` here, should there? The return is a callback to cancel the effect.
```suggestion
		setImageUrl( undefined );
```